### ISSUE TITLE
Linking in Emscripten >= 3.1.52 doesn't work

### ISSUE THREAD

**Issue by bz-next** [mosra/magnum#633]
I think the issue is analogous to https://github.com/emscripten-core/emscripten/issues/21128

The cube example builds with emscripten 3.1.51, but on 3.1.52 and above, I get the following error during linking of `Release/bin/magnum-primitives.js`

```
[100%] Linking CXX executable Release/bin/magnum-primitives.js
Release/lib/MagnumSdl2Application.bc:1:2: error: expected integer
    1 | !<arch>
      |  ^
1 error generated.
em++: error: '/home/user/emsdk/upstream/bin/clang++ -target wasm32-unknown-emscripten -fignore-exceptions -fvisibility=default --sysroot=/home/user/emsdk/upstream/emscripten/cache/sysroot -DEMSCRIPTEN -Xclang -iwithsysroot/include/fakesdl -Xclang -iwithsysroot/include/compat -DNDEBUG -O3 -flto -O3 -DNDEBUG -O3 -flto Release/lib/MagnumSdl2Application.bc -c -o /tmp/emscripten_temp_zy0k2l8p/MagnumSdl2Application_0.o' failed (returned 1)
make[2]: *** [CMakeFiles/magnum-primitives.dir/build.make:112: Release/bin/magnum-primitives.js] Error 1
make[1]: *** [CMakeFiles/Makefile2:762: CMakeFiles/magnum-primitives.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by mosra** [comment]
Ugh, this one is annoying. I don't see why would they want to change a behavior that worked well for a decade...

But what can I do :sweat_smile: I'll have to update all my Emscripten toolchains and builds [to not use `*.bc` but `*.a` instead](https://github.com/mosra/toolchains/blob/6a9e0821b4ccd578feee1f30cef245cbd36cdd38/modules/Platform/Emscripten.cmake#L22) and ensure everything reflects that. It'll take a while for this to propagate to all repos, and I don't really have time to do such a sweeping change at the moment. Is it fine if this waits a bit, can you stay on 3.1.51 for now? (It'll bite me as soon as I have 3.1.52 [in Arch repos](https://archlinux.org/packages/extra/x86_64/emscripten/), so it *will* get fixed eventually, just not now.)

**Comment by mosra** [comment]
Oh yeah, thanks for that! 

Emscripten breaks _a lot_ and very often, so this is something to be aware of. Basically every other upgrade breaks something, last time it was with 3.1.49 in 8048b65964e1bb171b637c6edd2699b43f45e2ea. So ideally on your side if possible you pin to some particular Emscripten version and upgrade only when you feel like you need to, or when you have a lot of extra time on your hands :D

**Comment by mosra** [comment]
This is fixed with mosra/toolchains@7c5e5047b3afd3f7a8292d8458d3c347ce265078, update of the submodule in Magnum repos will follow soon as well.
